index.html

```diff
@@ -4314,14 +4347,113 @@ <h3>Interactability</h3>
 </ol>
 </section> <!-- /Interactability -->
 
+<section>
+<h3 id=shadow-root>Shadow Roots</h3>
+<p>A <dfn data-lt="shadow roots">shadow root</dfn>
+  is an abstraction used to identify a <a>shadow root</a>
+  when it is transported via the <a href="#protocol">protocol</a>,
+  between <a>remote</a>- and <a>local</a> ends.
+ 
+ <p>The <dfn>shadow root identifier</dfn> is the string constant
+  "<code>shadow-6066-11e4-a52e-4f735466cecf</code>".
+ 
+ <p>Each <a>shadow root</a> has an associated <dfn>shadow root
+  reference</dfn> that uniquely identifies the <a>element</a> across
+  all <a>browsing contexts</a>.  The <a>shadow root reference</a> for
+  every <a>shadow root</a> representing the same <a>shadow root</a> must be the
+  same. It must be a string, and should be the result of <a>generating
+  a UUID</a>.
+ 
+ <p>An ECMAScript <a>Object</a> <dfn>represents a shadow root</dfn>
+  if it has a <a>shadow root identifier</a> <a>own property</a>.
+ 
+ <p>Each <a>browsing context</a> has an associated <dfn>list of
+  known shadow roots</dfn>.
+  When the <a>browsing context</a> is <a>discarded</a>,
+  the <a>list of known shadow roots</a> is discarded along with it.
+
+  <p>To <dfn>get a known shadow root</dfn> with
+    argument <var>reference</var>, run the following steps:
+   
+   <ol>
+    <li>Let <var>shadow</var> be the item in the <a>current browsing
+    context</a>’s <a>list of known shadow roots</a> for which the <a>shadow
+    root reference</a> is equal to <var>reference</var>, if such a
+    shadow root exists. Otherwise return <a>error</a> with <a>error
+    code</a> <a>no such element</a>.
+    <li>If <var>shadow</var> <a>is detached</a>, return
+    <a>error</a> with <a>error code</a>
+    <a>detached shadow root</a>.
+    <li>Return <a>success</a> with <var>shadow</var>.
+   </ol>
+   
+   <p>To <dfn>get a known connected shadow root</dfn> with
+   argument <var>reference</var>, run the following steps:
+   <ol>
+     <li>Let <var>shadow</var> be the result of <a>trying</a> to <a>get
+     a known shadow root</a> with argument <var>reference</var>.
+     <li>If <var>shadow</var> <a>is detached</a>
+     return <a>error</a> with error code <a>detached shadow root</a>.
```

Comment:
This check is already done by `get a known shadow root`, so these two functions are effectively the same.

Reply:
Will remove redundant function.